langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -269,6 +279,35 @@ private static ChatMessage prepareUserMessage(Method method, Object[] args) {
         throw illegalConfiguration("For methods with multiple parameters, each parameter must be annotated with @V, @UserMessage, @UserName or @MemoryId");
     }
 
+    private static String getPromptText(Method method, String type, String resource, String[] value, String delimiter) {
+        String systemMessageTemplate;
+        if (!resource.trim().isEmpty()) {
+            systemMessageTemplate = getResourceText(method.getDeclaringClass(), resource);
+            if (systemMessageTemplate == null) {
+                throw illegalConfiguration("@%sMessage's resource '%s' not found", type, resource);
+            }
+        } else {
+            systemMessageTemplate = String.join(delimiter, value);
+        }
+        if (systemMessageTemplate.trim().isEmpty()) {
+            throw illegalConfiguration("@%sMessage's template cannot be empty", type);
+        }
+        return systemMessageTemplate;
+    }
+
+    public static String getResourceText(Class<?> clazz, String name) {
+        return getText(clazz.getResourceAsStream(name));
+    }
+
+    public static String getText(InputStream inputStream) {
+        if (inputStream == null) {
+            return null;
+        }
+        try (Scanner scanner = new Scanner(inputStream); //
+                Scanner s = scanner.useDelimiter("\\A")) {
+            return s.hasNext() ? s.next() : "";
```

Comment:
should it return `null` instead of an empty string?

Reply:
When I read DefaultAiServices.getPromptText I would say that "" makes more sense: it will respond in the same way as when the "String[] value" is 'empty': with a message "template cannot be empty", indicating the resource was found but it was empty.